R/derive_vars_query.R

```diff
@@ -69,6 +69,41 @@ derive_vars_query <- function(dataset, dataset_queries) {
     optional = FALSE
   )
 
+  # check optionality of TERMNUM or TERMCHAR based on SRCVAR type
+  srcvar_types <- unique(vapply(dataset[dataset_queries$SRCVAR], typeof, character(1)))
+  if (srvar_types == "character") {
```

Comment:
@millerg23 @bundfussr Before I draft up tests and additional documentation, was this the intuition behind #2264 ?